Userland/Libraries/LibWeb/HTML/MimeTypeArray.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2023, Luke Wilde <lukew@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/HTML/MimeTypeArray.h>
+#include <LibWeb/HTML/Scripting/Environments.h>
+#include <LibWeb/HTML/Window.h>
+#include <LibWeb/Page/Page.h>
+
+namespace Web::HTML {
+
+MimeTypeArray::MimeTypeArray(JS::Realm& realm)
+    : Bindings::LegacyPlatformObject(realm)
+{
+}
+
+MimeTypeArray::~MimeTypeArray() = default;
+
+JS::ThrowCompletionOr<void> MimeTypeArray::initialize(JS::Realm& realm)
+{
+    MUST_OR_THROW_OOM(Base::initialize(realm));
+    set_prototype(&Bindings::ensure_web_prototype<Bindings::MimeTypeArrayPrototype>(realm, "MimeTypeArray"));
+
+    return {};
+}
+
+// https://html.spec.whatwg.org/multipage/system-state.html#pdf-viewing-support:support-named-properties-2
+Vector<DeprecatedString> MimeTypeArray::supported_property_names() const
+{
+    // The MimeTypeArray interface supports named properties. If the user agent's PDF viewer supported is true, then they are the PDF viewer mime types. Otherwise, they are the empty list.
+    auto const& window = verify_cast<HTML::Window>(HTML::relevant_global_object(*this));
+    VERIFY(window.page());
+    if (!window.page()->pdf_viewer_supported())
+        return {};
+
+    // https://html.spec.whatwg.org/multipage/system-state.html#pdf-viewer-mime-types
+    static Vector<DeprecatedString> mime_types = {
+        "application/pdf"sv,
+        "text/pdf"sv,
+    };
+
+    return mime_types;
+}
+
+// https://html.spec.whatwg.org/multipage/system-state.html#pdf-viewing-support:supports-indexed-properties-2
+bool MimeTypeArray::is_supported_property_index(u32 index) const
+{
+    // The MimeTypeArray interface supports indexed properties. The supported property indices are the indices of this's relevant global object's PDF viewer mime type objects.
+    auto& window = verify_cast<HTML::Window>(HTML::relevant_global_object(*this));
+    return index < window.pdf_viewer_mime_type_objects().size();
+}
+
+// https://html.spec.whatwg.org/multipage/system-state.html#dom-mimetypearray-length
+size_t MimeTypeArray::length() const
+{
+    // https://html.spec.whatwg.org/multipage/system-state.html#dom-plugin-length
```

Comment:
`// The MimeTypeArray interface's length getter steps are to return this's relevant global object's PDF viewer mime type objects's size.`